src/main/scala/redis/RedisPool.scala

```diff
@@ -95,3 +158,118 @@ case class RedisClientMasterSlaves(master: RedisServer,
 
   def redisConnection: ActorRef = masterClient.redisConnection
 }
+
+
+case class SentinelMonitoredRedisClientMasterSlaves(sentinels: Seq[(String, Int)] = Seq(("localhost", 26379)),
+                                         master: String)
+                                  (implicit _system:ActorSystem)
+  extends SentinelMonitored(_system) with ActorRequest  with  RedisCommands with Transactions {
+
+  val masterClient: RedisClient = withMasterAddr((ip, port) => {
+    new RedisClient(ip, port, name = "SMRedisClient")
+  })
+
+  val slavesClients:RedisClientMutablePool = withSlavesAddr( slavesHostPort => {
+    val slaves =  slavesHostPort.map{ case (ip, port) => {
+      new RedisServer(ip, port)
+    }}
+    new RedisClientMutablePool(slaves, name = "SMRedisClient")
+  })
+
+
+  val onNewSlave = (ip: String, port: Int)  => {
+      log.info(s"onNewSlave ${ip}:${port}")
+      slavesClients.addServer(RedisServer(ip,port))
+    }
+
+    val onSlaveDown = (ip: String, port: Int)  => {
+      log.info(s"onSlaveDown ${ip}:${port}")
+      slavesClients.removeServer(RedisServer(ip,port))
+    }
+
+    val onMasterChange = (ip: String, port: Int) => {
+      log.info(s"onMasterChange ${ip}:${port}")
+      masterClient.reconnect(ip, port)
+    }
+
+    /**
+      * Disconnect from the server (stop the actors)
+      */
+    def stop() = {
+      masterClient.stop()
+      slavesClients.stop()
+      sentinelClients.values.foreach(_.stop())
+    }
+
+
+
+
+  def redisConnection: ActorRef = masterClient.redisConnection
+
+  override def send[T](redisCommand: RedisCommand[_ <: RedisReply, T]): Future[T] = {
+    if (redisCommand.isMasterOnly || slavesClients.getConnectionsActive.isEmpty) {
```

Comment:
yeah i think we can be even smarter trying to mix in the `getNextConnection` in `RoundRobinPoolRequest`

The ideal would be to find the `next` active connection in the `RoundRobinPoolRequest`
If we find it, then we have a slave where we can send the command. Otherwise we can send the command to the master.